### PR TITLE
.when with weekdays and prefer_past

### PR DESCRIPTION
**Pull request by alxwrd**
This PR adds some more tests around `.when()` and the `prefer_past` paramenter.

I did change the test `test_when_past`. If you prefer the old test, I can change it back. Also, I added `test_when_future`, which is the opposite of `test_when_past`.

The other 2 tests: `test_when_past_day_name` and `test_when_future_day_name` (failing) to  demonstrate #146.

### CODE REVIEW

**Comment by alxwrd** [comment]
The last commit fixes #146 by allowing all three of dateparsers options of 'current_period', 'future', and 'past' to be accessible.

File: maya/core.py [hunk 1]
```diff
@@ -675,7 +675,7 @@ def now():
     return MayaDT(epoch=epoch)
 
 
-def when(string, timezone='UTC', prefer_past=False):
+def when(string, timezone='UTC', prefer_dates_from='current_period'):
```
Comment (timofurrer) [inline maya/core.py]: Can you document the possible values of `prefer_dates_from` in the docstring of `maya.when`?
Maybe also reference the dateparser documentation.

File: tests/test_maya.py [hunk 2]
```diff
@@ -191,16 +192,40 @@ def test_parse(string, kwds, expected):
...
-    future_date = maya.when(next_month)
-    past_date = maya.when(next_month, prefer_past=True)
-    assert future_date.year == this_year
-    if next_month == '1':
-        assert past_date.year == this_year
-    else:
-        assert past_date.year == last_year
+    two_days_away = maya.now().add(days=2)
+
+    past_date = maya.when(two_days_away.slang_date(), prefer_past=True)
```
Comment (timofurrer) [inline tests/test_maya.py]: Still uses the old `prefer_past` argument.

File: tests/test_maya.py [hunk 3]
```diff
@@ -191,16 +192,40 @@ def test_parse(string, kwds, expected):
...
+
+    assert past_date < maya.now()
+
+
+@pytest.mark.usefixtures("frozen_now")
+def test_when_future():
+    two_days_away = maya.now().add(days=2)
+
+    future_date = maya.when(two_days_away.slang_date(), prefer_past=False)
```
Comment (timofurrer) [inline tests/test_maya.py]: Still uses the old `prefer_past` argument.

File: tests/test_maya.py [hunk 4]
```diff
@@ -191,16 +192,40 @@ def test_parse(string, kwds, expected):
...
+
+    assert future_date > maya.now()
+
+
+@pytest.mark.usefixtures("frozen_now")
+def test_when_past_day_name():
+    two_days_away = maya.now().add(days=2)
+
+    past_date = maya.when(
+            calendar.day_name[two_days_away.weekday], prefer_past=True)
```
Comment (timofurrer) [inline tests/test_maya.py]: Still uses the old `prefer_past` argument.

File: tests/test_maya.py [hunk 5]
```diff
@@ -191,16 +192,40 @@ def test_parse(string, kwds, expected):
...
+
+    assert past_date < maya.now()
+
+
+@pytest.mark.usefixtures("frozen_now")
+def test_when_future_day_name():
+    two_days_away = maya.now().add(days=2)
+
+    future_date = maya.when(
+            calendar.day_name[two_days_away.weekday], prefer_past=False)
```
Comment (timofurrer) [inline tests/test_maya.py]: Still uses the old `prefer_past` argument.

File: maya/core.py (hunk 1 above)
Comment (alxwrd) [reply maya/core.py]: oops, completely forgot the docstring! Should be updated now.

**Pull request merged**